internal/host/refresh_status_preprocessor.go

```diff
@@ -93,6 +94,16 @@ func (r *refreshPreprocessor) preprocess(ctx context.Context, c *validationConte
 			message = validationDisabledByConfiguration
 			conditions[v.id.String()] = true
 		} else {
+			// We want to differentiate between CMN and UMN. If it's UMN, we will execute a IsMtuOKAllInterfaces validation; otherwise, IsMtuOKInMachineNetwork validation will be executed.
```

Comment:
Shouldn't those check be part of the validator function itself (and return early a ValidationSuccessSuppressOutput if the test is not relevant) ?

I'm not familiar with this part of the code, but it would keep this function generic

Reply:
Agree. Thanks!